src/query/functions-v2/src/scalars/hash.rs

```diff
@@ -0,0 +1,309 @@
+                        let mut hasher = CityHasher64::with_seed(l);
+                        DFHash::hash(&val, &mut hasher);
+                        output.push(hasher.finish());
+                        Ok(())
+                    },
+                ),
+                );
+                registry.register_passthrough_nullable_2_arg::<TimestampType, NumberType<R>, NumberType<u64>, _, _>(
+                "city64withseed",
+                FunctionProperty::default(),
+                |_, _| FunctionDomain::MayThrow,
+                vectorize_with_builder_2_arg::<TimestampType, NumberType<R>, NumberType<u64>>(
+                    |val, l, output, _| {
+                        let l: u64 = l.as_();
+                        let mut hasher = CityHasher64::with_seed(l);
+                        DFHash::hash(&val, &mut hasher);
+                        output.push(hasher.finish());
+                        Ok(())
+                    },
+                ),
+                );
+            }
+        })
+    }
+}
+
+pub const MAX_NUMERICS_TYPES: &[NumberDataType; 3] = &[
+    NumberDataType::Int64,
+    NumberDataType::UInt64,
+    NumberDataType::Float64,
+];
+
+struct CityHasher64 {
+    seed: u64,
+    value: u64,
+}
+
+impl CityHasher64 {
+    fn with_seed(s: u64) -> Self {
+        Self { seed: s, value: 0 }
+    }
+}
+
+impl Hasher for CityHasher64 {
+    fn finish(&self) -> u64 {
+        self.value
+    }
+
+    fn write(&mut self, bytes: &[u8]) {
+        self.value = cityhash64_with_seed(bytes, self.seed);
+    }
+}
+
+pub trait DFHash {
+    fn hash<H: Hasher>(&self, state: &mut H);
+}
+
+macro_rules! integer_impl {
+    ([], $( { $S: ident} ),*) => {
+        $(
+            impl DFHash for $S {
+                #[inline]
+                fn hash<H: Hasher>(&self, state: &mut H) {
+                    Hash::hash(self, state);
+                }
+            }
+        )*
+    }
+}
+
+for_all_integer_types! { integer_impl }
+
+impl DFHash for F32 {
+    #[inline]
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        let u = self.to_bits();
+        Hash::hash(&u, state);
+    }
+}
+
+impl DFHash for F64 {
+    #[inline]
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        let u = self.to_bits();
+        Hash::hash(&u, state);
+    }
+}
+
+impl<'a> DFHash for &'a [u8] {
+    #[inline]
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        Hash::hash_slice(self, state);
+    }
+}
+
+impl DFHash for Scalar {
+    #[inline]
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        match self {
+            Scalar::Number(NumberScalar::Int64(v)) => DFHash::hash(v, state),
+            Scalar::Number(NumberScalar::UInt64(v)) => DFHash::hash(v, state),
+            Scalar::Number(NumberScalar::Float64(v)) => DFHash::hash(v, state),
+            Scalar::String(vals) => {
+                for v in vals {
+                    DFHash::hash(v, state);
+                }
+            }
+            _ => {}
```

Comment:
what if the other is uint8 ... ?

`with_number_mapped_type` could be used there.